### PR TITLE
Adding delete + labels modals to vmi list

### PR DESCRIPTION
**Pull request by metalice**
## 📝 Description

> Add a brief description

## 🎥 Demo

image

image

### CODE REVIEW

**Review by hstastna — CHANGES_REQUESTED** [review]
All the labels in the Kebab menu won't be translated like this. I'd suggest to add the `t` function to all of them:
translate

The modal for editing labels could work a bit better. I have to click on _Submit_ button twice (or somewhere else in the screen for the first time), when adding/editing some label, it's annoying. It also adds an unnecessary empty line, when clicking on Submit (or anywhere else) for the first time. When removing some label or canceling this action, I have to click on Submit only once. I'd suggest to look into the implementation of _LabelsModal_ you use here. Maybe we can do something about that, maybe not, I'm not sure.
hello

Otherwise the code looks great and deleting VMIs or editing their labels works! 👍🏽

File: src/views/virtualmachinesinstance/actions/hooks/useVirtualMachineInstanceActionsProvider.tsx [hunk 1]
```diff
@@ -35,23 +42,77 @@ const useVirtualMachineInstanceActionsProvider: UseVirtualMachineInstanceActions
       {
         id: 'edit-labels',
         label: 'Edit Labels',
```
[the remark this answers is not in the document]
Comment (metalice) [reply src/views/virtualmachinesinstance/actions/hooks/useVirtualMachineInstanceActionsProvider.tsx:44]: great comment!

File: src/views/virtualmachinesinstance/actions/hooks/useVirtualMachineInstanceActionsProvider.tsx [hunk 2]
```diff
@@ -35,23 +42,77 @@ const useVirtualMachineInstanceActionsProvider: UseVirtualMachineInstanceActions
...
+            />
+          )),
       },
       {
         id: 'edit-annotations',
         label: 'Edit Annotations',
-        // waiting for modal launcher
-        cta: () => console.log('helloWorld'),
+        cta: () =>
+          // createModal(({ isOpen, onClose }) => (
```
[the remark this answers is not in the document]
Comment (metalice) [reply src/views/virtualmachinesinstance/actions/hooks/useVirtualMachineInstanceActionsProvider.tsx:71]: indeed this PR was waited for the annotation label to be merged first and this is why it's in the comment. now when the annotation modal has been merged this pr can be edited and merged.

**Comment by metalice** [comment]
> All the labels in the Kebab menu won't be translated like this. I'd suggest to add the `t` function to all of them: translate
> 
> The modal for editing labels could work a bit better. I have to click on _Submit_ button twice (or somewhere else in the screen for the first time), when adding/editing some label, it's annoying. It also adds an unnecessary empty line, when clicking on Submit (or anywhere else) for the first time. When removing some label or canceling this action, I have to click on Submit only once. I'd suggest to look into the implementation of _LabelsModal_ you use here. Maybe we can do something about that, maybe not, I'm not sure. hello
> 
> Otherwise the code looks great and deleting VMIs or editing their labels works! 👍🏽

This PR didn't create the modals only consumed them, if those modals act "weird" we should tackle it, but in a follow-up PR. Thank you for bringing it to my attention.

**Pull request merged**